File: Python/Bucles_for/Avanzado/Ejercicio_05.py
# Lectura de la matriz A desde la entrada estándar
matriz_a = [[int(input(f"Ingrese el elemento A[{i}][{j}]: ")) for j in range(4)] for i in range(4)]

# Lectura de la matriz B desde la entrada estándar
matriz_b = [[int(input(f"Ingrese el elemento B[{i}][{j}]: ")) for j in range(4)] for i in range(4)]

# Inicialización de la matriz resultado
matriz_resultado = [[0] * 4 for _ in range(4)]

# Multiplicación de matrices
for i in range(4):
    for j in range(4):
        for k in range(4):
            matriz_resultado[i][j] += matriz_a[i][k] * matriz_b[k][j]

# Mostrar la matriz resultado
print("Matriz Resultado (A * B):")
for i in range(4):
    for j in range(4):
        print(matriz_resultado[i][j], end=" ")
    print()

File: Java/src/main/java/org/sisuns/bucles_for/avanzado/Ejercicio_05.java
package org.sisuns.bucles_for.avanzado;

import java.util.Scanner;

public class Ejercicio_05 {
    public static void main(String[] args) {
        // Declaración de variables
        int[][] matrizA = new int[4][4];
        int[][] matrizB = new int[4][4];
        int[][] matrizResultado = new int[4][4];
        int suma;

        // Lectura de la matriz A desde la entrada estándar
        System.out.println("Ingrese los elementos de la matriz A (4x4):");
        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                System.out.print("Ingrese el elemento A[" + i + "][" + j + "]: ");
                matrizA[i][j] = scanner.nextInt();
            }
        }

        // Lectura de la matriz B desde la entrada estándar
        System.out.println("Ingrese los elementos de la matriz B (4x4):");
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                System.out.print("Ingrese el elemento B[" + i + "][" + j + "]: ");
                matrizB[i][j] = scanner.nextInt();
            }
        }

        // Multiplicación de matrices
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                suma = 0;
                for (int k = 0; k < 4; k++) {
                    suma += matrizA[i][k] * matrizB[k][j];
                }
                matrizResultado[i][j] = suma;
            }
        }

        // Mostrar la matriz resultado
        System.out.println("Matriz Resultado (A * B):");
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                System.out.print(matrizResultado[i][j] + " ");
            }
            System.out.println();
        }
    }
}
